spectrocloud/cluster_common_fields.go

```diff
@@ -23,10 +23,12 @@ func readCommonFields(c *client.V1Client, d *schema.ResourceData, cluster *model
 		return diag.FromErr(err), true
 	}
 	// When the current repave state is pending, we set the review_repave_state to Pending, For indicate the system change.
-	if err := d.Set("review_repave_state", cluster.Status.Repave.State); err != nil {
-		return diag.FromErr(err), true
+	if _, ok := d.GetOk("review_repave_state"); ok {
+		// We are adding this check to handle virtual cluster scenario. virtual cluster doesn't have support for `review_repave_state`
+		if err := d.Set("review_repave_state", cluster.Status.Repave.State); err != nil {
```

Comment:
@SivaanandM could you please add a unit test scenario to cover this branch?

Reply:
its already added as part of cluster